helpers.py

```diff
@@ -42,7 +42,7 @@ def __exit__(self, *args):
             
 def cudify(x):
     if use_cuda:
-        return x.cuda(async=True)
+        return x.cuda(non_blocking=True)
```

Comment:
This looks reasonable, although it would be helpful if you could attach log results from tests demonstrating the same results on the networks provided, with the same installation setup.